File: python/HardDrive.py
from Komponen import Komponen
class HardDrive(Komponen) :
    def __init__(self, kapasitasGB, tipeDrive, merk, nama):
        super().__init__(merk, nama)
        
        self.__kapasitasGB = kapasitasGB
        self.__tipeDrive = tipeDrive
        
    def get_kapasitasGB(self):
        return self.__kapasitasGB
    def get_tipeDrive(self):
        return self.__tipeDrive
    
    # Setter methods
    def set_kapasitasGB(self, kapasitasGB):
        self.__kapasitasGB = kapasitasGB
    def set_tipeDrive(self, tipeDrive):
        self.__tipeDrive = tipeDrive
        

File: cpp/HardDrive.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;
class Harddrive : public Komponen {
private:
    int kapasitasGB;
    string tipeDrive;

public:
    Harddrive()

    {
    }
    Harddrive(int kapasitasGB, string tipeDrive, string merk, string nama)
        : Komponen(merk, nama) {
        this->kapasitasGB = kapasitasGB;
        this->tipeDrive = tipeDrive;
    }
    void setKapasitasGB(int kapasitasGB) {
        this->kapasitasGB = kapasitasGB;
    }
    void setTipeDrive(float tipeDrive) {
        this->tipeDrive = tipeDrive;
    }
    int getKapasitasGB() {
        return this->kapasitasGB;
    }
    string getTipeDrive() {
        return this->tipeDrive;
    }
    ~Harddrive() {
    }
};